cadc-vosi/src/main/java/ca/nrc/cadc/vosi/avail/CheckDataSource.java

```diff
@@ -141,13 +148,26 @@ public void check()
             {
                 this.dataSource = DBUtil.findJNDIDataSource(dataSourceName);
             }
+            
             con = dataSource.getConnection();
+            if (this.rollback)
+            {
+                con.setAutoCommit(false);
+            }
+            
             st = con.createStatement();
             if (expectResults)
             {
                 log.debug("test for results");
-                rs = st.executeQuery(testSQL);
-                rs.next(); // just check the result set, but don't care if there are any rows
+                if (testSQL.trim().toLowerCase().startsWith("select "))
+                {
+                    rs = st.executeQuery(testSQL);
+                    rs.next(); // just check the result set, but don't care if there are any rows
+                }
+                else
+                {
+                    int count = st.executeUpdate(testSQL);
```

Comment:
No need to capture it. I put it in when I was testing the code. I can remove 'int count = '.